datastore/in_memory/member.go

```diff
@@ -3,33 +3,67 @@ package in_memory
 import (
 	"errors"
 	"memberserver/api/models"
+	"sort"
 )
 
 var Members = map[string]models.Member{}
 
 func (i In_memory) GetTiers() []models.Tier {
-	return []models.Tier{}
+	return i.Tiers
 }
 func (i In_memory) GetMembers() []models.Member {
-	return []models.Member{}
+	return MemberMapToSlice(i.Members)
 }
 func (i In_memory) GetMemberByEmail(email string) (models.Member, error) {
-	for _, k := range testUsers {
+	for _, k := range i.Members {
 		if k.Email == email {
 			return k, nil
 		}
 	}
 	return models.Member{}, errors.New("error getting user: not found")
 }
 func (i In_memory) AssignRFID(email string, rfid string) (models.Member, error) {
-	return models.Member{}, nil
+	if len(rfid) == 0 {
+		return models.Member{}, errors.New("not a valid rfid")
+	}
+
+	for _, member := range i.Members {
+		if member.Email != email {
+			continue
+		}
+		return member, nil
+	}
+	return models.Member{}, errors.New("user not found")
 }
 func (i In_memory) AddNewMember(newMember models.Member) (models.Member, error) {
-	return models.Member{}, nil
+	return newMember, nil
 }
 func (i In_memory) AddMembers(members []models.Member) error {
 	return nil
 }
 func (i In_memory) GetMembersWithCredit() []models.Member {
 	return []models.Member{}
 }
+func (i In_memory) ProcessMember(newMember models.Member) error {
+	return nil
+}
+
+func MemberMapToSlice(m map[string]models.Member) []models.Member {
+	var members []models.Member
+
+	for _, member := range m {
+		members = append(members, member)
+	}
+
+	sort.Sort(ByID(members))
```

Comment:
why are you sorting by Id for member?

Reply:
it's because maps are unordered in golang.
```goalng 
func MemberMapToSlice(m map[string]models.Member) []models.Member 
```

This function takes a map as an argument and translates it to a slice.
I want the results to be the same every time.  So, I sort it.

This is a helper function that helps the tests simulate what's happening in the DB.

Reply:
So, this is mainly used for testing? if so, put it in the test file

Reply:
This file implements a MemberStore.
This is an in_memory implementation of it.

the sorting is required to have it mimic the other implementations.  So, I don't want to put it in a test file.  Mainly because the tests that use it aren't aware that it's happening.